### DISCUSSION TITLE
Plans to introduce "Service" along with "Product" and "Asset"

### DISCUSSION THREAD

**Discussion by javac9** [opencrx/opencrx#74] [Q&A]
I guess it would make much sense to introduce "Service" as a new type of things that may be for sale.

**Comment by wfro64** [comment]
Let's suppose we would add the new type 'Service' as subclass of 'Product' or 'AbstractProduct'. What would be the expected new features for this new type? Maybe some examples would help.

**Comment by javac9** [comment]
I find it more a "semantic" issue than a "practical" one at this moment. English language lacks words that would seamlessly denote both (tangible) products and services but here I found some interesting proposals (of which I find only one applicable) - **deliverable/s**
[https://english.stackexchange.com/questions/160428/single-word-for-product-or-service](https://english.stackexchange.com/questions/160428/single-word-for-product-or-service)

Sure, one can argue that being that picky and changing it to this would be an overkill - but I'm pretty sure it would look much better. I will also try to come up with some features for "service" that might make it worth the change in your eyes.

**Comment by javac9** [comment]
I'm getting back to this. I gave it a bit of thought and I think new features (the ones that would clearly help distinguish "services" from current "products") could be something like these:

- a service may be provided / carried out online / via Internet. Example: online language classes/tutoring. Here I can see various tools involved when booking (Skype, Zoom, Google Duo, MS Teams, etc.), then a field to store join link, passcode, etc.
- such online services may be easily/frequently cancelled/postponed so maybe support that in some way
- another pretty probable use-case would be providing tech support via phone, email, or similar
- maybe support keeping notes/minute-book, on a per-event basis?
- I don't know if OpenCrx supports any kind of creating tickets but this would be awesome to have in combination with the above proposed use-case with tech support.

I may add more items later if I remembers something.

**Comment by javac9** [comment]
Any feedback on this? Btw, I suggest to label it with `Ideas`

**Comment by wfro64** [comment]
There are no plans to have a "Service" type in addition to the existing "Product". All use cases you mention can be realized with the existing functions:
* map customer and its addresses (email , postal, chat, ...) to Contact
* map online service types to Products and Prices
* map subscriptions to SalesContracts / SalesContractPositions
* map customer interactions (subscribe, unsubscribe, cancel, ...) to Activities
* map ticket management to Activities, ActivityCreators, ActivityGroups tickets
